lib/mods/replace-constructors.js

```diff
@@ -71,8 +16,10 @@ function replaceGlobalConstructors(node) {
   if (nodeTypes.isNewExpression(node) &&
       isGlobal(node)) {
     const identifier = node.getExpression();
-    identifier.replaceWithText(getOriginal(
-        node.getType().getSymbol().getName()));
+    const mod = node.getType().getSymbol().getName();
+    const original = getOriginal(mod, 'constructor');
+    identifier.replaceWithText(original);
+    updateImports(mod, original);
```

Comment:
The use of hidden global state, i.e. `IMPORTS` inside the helpers.js file, is not a great pattern here. However, I can understand the motivation; it would be annoying to have to add the imports as you go, because then you'd have to see if any were previously added.

Instead, what about an architecture where each mod (e.g. `replaceGlobalConstructors`) accepts and updates an `imports` object. Then you create that object inside `rewriter.js`, pass it to each of the mods, and then pass it to `addAllImports` in the end.

Reply:
Yeah I thought a bit about what the best pattern for this would be, I agree that hidden global state should be cautioned against (and didn't think of this way to do it at the time). I'll switch to this.